x-pack/plugins/security_solution/public/common/components/super_date_picker/index.tsx

```diff
@@ -312,19 +321,35 @@ export const makeMapStateToProps = () => {
   const getStartSelector = startSelector();
   const getToStrSelector = toStrSelector();
   return (state: State, { id }: OwnProps) => {
-    const inputsRange: InputsRange = getOr({}, `inputs.${id}`, state);
+    if (id === 'socTrends') {
+      const inputsRange: InputsRangeTimeOnly = getOr({}, `inputs.${id}`, state);
+      return {
+        duration: getDurationSelector(inputsRange),
+        end: getEndSelector(inputsRange),
+        fromStr: getFromStrSelector(inputsRange),
+        kind: getKindSelector(inputsRange),
+        policy: getPolicySelector(inputsRange),
+        start: getStartSelector(inputsRange),
+        toStr: getToStrSelector(inputsRange),
+        isLoading: false,
+        // needs to be defined as undefined ¯\_(ツ)_/¯
+        kqlQuery: undefined,
+        queries: [],
+      };
+    }
```

Comment:
This `if (id === 'socTrends')` looks a bit too specific for a common/generic module, if there's any change we'll have to remember to come here to change this logic. 
What if we create a type guard, so we can define the inputs that have `queries` and the ones that don't have them, in a generic way? 
We could do:
```
export type Inputs = InputsRange | InputsRangeTimeOnly;

export function isQueryInput(inputs: Inputs): inputs is InputsRange {
  if ('queries' in inputs) {
    return true;
  }
  return false;
}
```
in the `common/store/inputs/model.ts`, and use them here:
```
 return (state: State, { id }: OwnProps) => {
    const inputsRange: Inputs = getOr({}, `inputs.${id}`, state);
    return {
      duration: getDurationSelector(inputsRange),
      end: getEndSelector(inputsRange),
      fromStr: getFromStrSelector(inputsRange),
      kind: getKindSelector(inputsRange),
      policy: getPolicySelector(inputsRange),
      start: getStartSelector(inputsRange),
      toStr: getToStrSelector(inputsRange),
      isLoading: false,
      ...(isQueryInput(inputsRange) && {
        isLoading: getIsLoadingSelector(inputsRange),
        kqlQuery: getKqlQuerySelector(inputsRange),
        queries: getQueriesSelector(state, id),
      }),
    };
  };
```

Reply:
Awesome. I had to add a check alongside isQueryInput for `(id === 'timeline' || id === 'global')`, but it works now. thank you!